### PR TITLE
Bin-Sort in Python

### LINKED ISSUE #433: Bin-Sort in Python
This is an open issue for bin sorting algorithm implementation in python. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by vikumkbv**
**Hi!**  
Thanks for considering contributing to this ever-growing list of algorithm and data structure implementations.
Your contribution is valuable. Use the follow template for creating Pull Requests
---

**Fixes #433 **

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [&#10004; ] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [&#10004; ] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [&#10004; ] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [&#10004; ] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [&#10004; ] This pull request shall only be merged once the build passes.

-   [&#10004; ] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		
+	# concatenate the result 
+	k = 0
+	for i in range(slot_num): 
+		for j in range(len(arr[i])): 
+			x[k] = arr[i][j] 
+			k += 1
+	return x 
+
+# Driver Code
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:37]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	k = 0
+	for i in range(slot_num): 
+		for j in range(len(arr[i])): 
+			x[k] = arr[i][j] 
+			k += 1
+	return x 
+
+# Driver Code 
+x = [0.897, 0.565, 0.656, 
+	0.1234, 0.665, 0.3434]
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:39]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Python3 program to sort an array 
+# using bucket sort 
+def insertionSort(b): 
+	for i in range(1, len(b)): 
+		up = b[i]
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:5]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Python3 program to sort an array 
+# using bucket sort 
+def insertionSort(b): 
+	for i in range(1, len(b)): 
+		up = b[i] 
+		j = i - 1
+		while j >= 0 and b[j] > up: 
+			b[j + 1] = b[j] 
+			j -= 1
+		b[j + 1] = up
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:10]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+			b[j + 1] = b[j] 
+			j -= 1
+		b[j + 1] = up	 
+	return b	 
+			
+def bucketSort(x): 
+	arr = [] 
+	slot_num = 10 # 10 means 10 slots, each 
+				# slot's size is 0.1 
+	for i in range(slot_num):
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:17]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
+# Python3 program to sort an array
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:1]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+		arr.append([]) 
+		
+	# Put array elements in different buckets 
+	for j in x: 
+		index_b = int(slot_num * j) 
+		arr[index_b].append(j) 
+	
+	# Sort individual buckets 
+	for i in range(slot_num): 
+		arr[i] = insertionSort(arr[i])
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:27]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+	for i in range(slot_num): 
+		for j in range(len(arr[i])): 
+			x[k] = arr[i][j] 
+			k += 1
+	return x 
+
+# Driver Code 
+x = [0.897, 0.565, 0.656, 
+	0.1234, 0.665, 0.3434] 
+print("Sorted Array is")
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:40]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+	for i in range(1, len(b)): 
+		up = b[i] 
+		j = i - 1
+		while j >= 0 and b[j] > up: 
+			b[j + 1] = b[j] 
+			j -= 1
+		b[j + 1] = up	 
+	return b	 
+			
+def bucketSort(x):
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:13]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+		arr[index_b].append(j) 
+	
+	# Sort individual buckets 
+	for i in range(slot_num): 
+		arr[i] = insertionSort(arr[i]) 
+		
+	# concatenate the result 
+	k = 0
+	for i in range(slot_num): 
+		for j in range(len(arr[i])):
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:32]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+	return b	 
+			
+def bucketSort(x): 
+	arr = [] 
+	slot_num = 10 # 10 means 10 slots, each 
+				# slot's size is 0.1 
+	for i in range(slot_num): 
+		arr.append([]) 
+		
+	# Put array elements in different buckets
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:20]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+	for i in range(slot_num): 
+		arr[i] = insertionSort(arr[i]) 
+		
+	# concatenate the result 
+	k = 0
+	for i in range(slot_num): 
+		for j in range(len(arr[i])): 
+			x[k] = arr[i][j] 
+			k += 1
+	return x
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:35]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+	# concatenate the result 
+	k = 0
+	for i in range(slot_num): 
+		for j in range(len(arr[i])): 
+			x[k] = arr[i][j] 
+			k += 1
+	return x 
+
+# Driver Code 
+x = [0.897, 0.565, 0.656,
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:38]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 14]
```diff
@@ -0,0 +1,41 @@
+# Python3 program to sort an array 
+# using bucket sort 
+def insertionSort(b): 
+	for i in range(1, len(b)): 
+		up = b[i] 
+		j = i - 1
+		while j >= 0 and b[j] > up: 
+			b[j + 1] = b[j]
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:8]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+	# Put array elements in different buckets 
+	for j in x: 
+		index_b = int(slot_num * j) 
+		arr[index_b].append(j) 
+	
+	# Sort individual buckets 
+	for i in range(slot_num): 
+		arr[i] = insertionSort(arr[i]) 
+		
+	# concatenate the result
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:29]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+				# slot's size is 0.1 
+	for i in range(slot_num): 
+		arr.append([]) 
+		
+	# Put array elements in different buckets 
+	for j in x: 
+		index_b = int(slot_num * j) 
+		arr[index_b].append(j) 
+	
+	# Sort individual buckets
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:25]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+			j -= 1
+		b[j + 1] = up	 
+	return b	 
+			
+def bucketSort(x): 
+	arr = [] 
+	slot_num = 10 # 10 means 10 slots, each 
+				# slot's size is 0.1 
+	for i in range(slot_num): 
+		arr.append([])
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:18]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+# using bucket sort 
+def insertionSort(b): 
+	for i in range(1, len(b)): 
+		up = b[i] 
+		j = i - 1
+		while j >= 0 and b[j] > up: 
+			b[j + 1] = b[j] 
+			j -= 1
+		b[j + 1] = up	 
+	return b
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:11]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+		while j >= 0 and b[j] > up: 
+			b[j + 1] = b[j] 
+			j -= 1
+		b[j + 1] = up	 
+	return b	 
+			
+def bucketSort(x): 
+	arr = [] 
+	slot_num = 10 # 10 means 10 slots, each 
+				# slot's size is 0.1
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:16]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 20]
```diff
@@ -0,0 +1,41 @@
+# Python3 program to sort an array 
+# using bucket sort 
+def insertionSort(b):
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:3]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 21]
```diff
@@ -0,0 +1,41 @@
+# Python3 program to sort an array 
+# using bucket sort 
+def insertionSort(b): 
+	for i in range(1, len(b)):
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:4]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+	arr = [] 
+	slot_num = 10 # 10 means 10 slots, each 
+				# slot's size is 0.1 
+	for i in range(slot_num): 
+		arr.append([]) 
+		
+	# Put array elements in different buckets 
+	for j in x: 
+		index_b = int(slot_num * j) 
+		arr[index_b].append(j)
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:23]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+		j = i - 1
+		while j >= 0 and b[j] > up: 
+			b[j + 1] = b[j] 
+			j -= 1
+		b[j + 1] = up	 
+	return b	 
+			
+def bucketSort(x): 
+	arr = [] 
+	slot_num = 10 # 10 means 10 slots, each
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:15]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+def insertionSort(b): 
+	for i in range(1, len(b)): 
+		up = b[i] 
+		j = i - 1
+		while j >= 0 and b[j] > up: 
+			b[j + 1] = b[j] 
+			j -= 1
+		b[j + 1] = up	 
+	return b	 
+
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:12]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+def bucketSort(x): 
+	arr = [] 
+	slot_num = 10 # 10 means 10 slots, each 
+				# slot's size is 0.1 
+	for i in range(slot_num): 
+		arr.append([]) 
+		
+	# Put array elements in different buckets 
+	for j in x: 
+		index_b = int(slot_num * j)
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:22]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 26]
```diff
@@ -0,0 +1,40 @@
...
+def insertionSort(b):
+	for i in range(1, len(b)):
+		up = b[i]
+		j = i - 1
+		while j >= 0 and b[j] > up:
+			b[j + 1] = b[j]
+			j -= 1
+		b[j + 1] = up
+	return b
+
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:12]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 27]
```diff
@@ -0,0 +1,40 @@
...
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:12]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

File: algorithms/ar-binsrt/python3/binsrt.py [hunk 28]
```diff
@@ -0,0 +1,40 @@
...
+def bucketSort(x):
+	arr = []
+	slot_num = 10 # 10 means 10 slots, each
+				# slot's size is 0.1
+	for i in range(slot_num):
+		arr.append([])
+
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/python3/binsrt.py:19]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6317676)

**Review by Danushka96 — APPROVED** [review]

**Comment by RavinduSachintha** [comment]
@vikumkbv just add this contribution to the algorithm chart on the main readme file. don't change anything else. :smile:

**Comment by vikumkbv** [comment]
> 
> 
> @vikumkbv just add this contribution to the algorithm chart on the main readme file. don't change anything else. 😄

done all updated

**Comment by RavinduSachintha** [comment]
@all-contributors please add @vikumkbv for code and doc

**Comment by RavinduSachintha** [comment]
@vikumkbv thank you for your contribution. keep it up :+1: 
Make sure to star this repo :wink:

**Pull request merged**